api/src/ballots.py

```diff
@@ -46,11 +49,15 @@ def ballots():
   # TODO support historic data requests - run query on `rejected` table, otherwise run on main table
   historic = req.args.get('show_historic', False)
 
-  mydb = MySQLdb.connect(host=config['DATABASE']['host'],
-    user=config['DATABASE']['user'],
-    passwd=config['DATABASE']['passwd'],
-    db=config[state]['db'],
-    local_infile = 1)
+  try:
+    mydb = MySQLdb.connect(host=config['DATABASE']['host'],
+      user=config['DATABASE']['user'],
+      passwd=config['DATABASE']['passwd'],
+      db=config[state]['db'],
+      local_infile = 1)
+  except:
+    # if connection failed, then input state was not valid
+    return "404 bad request"
```

Comment:
This should be a 500 because it's an error regarding the internals of the server and not due to the request

Reply:
That line throws an exception if you give the wrong state because it's trying to connect to a state database that doesn't exist, so it's technically due to a bad request (I guess it could fail due to a failure of the connection too). 

Should I change this so before trying to connect it checks the config file to see if there is a key for that state (since that indicates the state isn't supported)?

Reply:
I see, I think putting it here makes it ambiguous then whether the connection failed due to an internal error of connecting to MySQL or due to bad user input - from my experience I've seen that catches here are usually for the MySQL connection (assuming good input) and then returning a 500. 

I think we can change this to a 500 and leave it for now, and time-permitting we can implement the config/database checks for 404's